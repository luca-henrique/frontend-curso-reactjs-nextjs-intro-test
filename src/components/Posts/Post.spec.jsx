import React from 'react';
import { render, screen } from '@testing-library/react';
import { Posts } from './index';

const mock = {
  posts: [
    {
      title: 'title 1',
      body: 'body 1',
      id: 1,
      cover: 'img/img.png',
    },
    {
      title: 'title 2',
      body: 'body 2',
      id: 2,
      cover: 'img/img.png',
    },
    {
      title: 'title 3',
      body: 'body 3',
      id: 3,
      cover: 'img/img.png',
    },
  ],
};

describe('Name of the group', () => {
  test('should render posts heading', () => {
    render(<Posts {...mock} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
  });

  test('should render posts img', () => {
    render(<Posts {...mock} />);

    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
  });

  test('should render posts body', () => {
    render(<Posts {...mock} />);

    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  test('should match snapshot', () => {
    const { container } = render(<Posts />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
