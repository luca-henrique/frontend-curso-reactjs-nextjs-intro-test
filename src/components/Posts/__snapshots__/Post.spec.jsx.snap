// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should match snapshot 1`] = `
<div
  class="posts"
/>
`;
